File: elefantinhos.rb

vezes = 1
qtd = 1
loop do
    sleep(1)
    if qtd == 1 
        puts("#{qtd} elefante incomoda muita gente!")
    else 
        if qtd % 2 == 0
            incomoda = ""
            i = 0
            for i in i..vezes do
                incomoda += "incomodam "
            end
            puts("#{qtd} elefantes #{incomoda}muito mais!")
        else 
            puts("#{qtd} elefantes incomodam muita gente!")
        end   
    end
    qtd+=1
    vezes+=1
end

File: elefantinhos.java
class Elefantinhos {
    public static void main(String[] args) {
    int vezes = 1, quantidade = 1;

        while(true){
            if(quantidade == 1){
                System.out.println(quantidade + " elefante incomoda muita gente!");
            }else{
                if(quantidade % 2 == 0){
                    String incomoda = "";
                    
                    for(int i = 0; i < vezes; i++){
                        incomoda += "incomodam ";
                    }
                    System.out.println(quantidade + " elefantes " + incomoda + "muito mais!");
                    }else{
                    System.out.println(quantidade + " elefantes incomodam muita gente!");
                }
            }
            quantidade++;
            vezes++;
        }
    }
}